android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFirebaseMessagingDelegate.kt

```diff
@@ -0,0 +1,89 @@
+package host.exp.exponent.fcm
+
+import android.content.Context
+import com.google.firebase.messaging.RemoteMessage
+import expo.modules.notifications.notifications.model.NotificationContent
+import expo.modules.notifications.notifications.model.NotificationRequest
+import expo.modules.notifications.notifications.model.triggers.FirebaseNotificationTrigger
+import expo.modules.notifications.service.delegates.FirebaseMessagingDelegate
+import host.exp.exponent.ABIVersion
+import host.exp.exponent.Constants
+import host.exp.exponent.analytics.EXL
+import host.exp.exponent.kernel.ExperienceId
+import host.exp.exponent.notifications.PushNotificationHelper
+import host.exp.exponent.notifications.model.ScopedNotificationRequest
+import host.exp.exponent.storage.ExponentDB
+import org.json.JSONException
+import org.json.JSONObject
+
+class ExpoFirebaseMessagingDelegate(context: Context) : FirebaseMessagingDelegate(context) {
+  override fun onNewToken(token: String) {
+    if (!Constants.FCM_ENABLED) {
+      return
+    }
+    super.onNewToken(token)
+    FcmRegistrationIntentService.registerForeground(context.applicationContext, token)
+  }
+
+  override fun onMessageReceived(remoteMessage: RemoteMessage) {
+    if (!Constants.FCM_ENABLED) {
+      return
+    }
+    val experienceDBObject = ExponentDB.experienceIdToExperienceSync(remoteMessage.data["experienceId"])
+    if (experienceDBObject != null) {
+      try {
+        val manifest = JSONObject(experienceDBObject.manifest)
+        val sdkVersion = ABIVersion.toNumber(manifest.getString("sdkVersion")) / 10000
```

Comment:
```suggestion
        val sdkVersion = ABIVersion.majorSdkVersion(manifest.getString("sdkVersion"))
```
Don't we have sth like this? 🤔 
I think we should have such mechanism & delegate all `SDK-version`-related things to that `ABIVersion` class or equivalent.

Reply:
I agree this is not the canonical way of comparing SDK versions, compare:

https://github.com/expo/expo/blob/2e6fbac4e7839c9f96fcf35900ec096fbc077a01/android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFirebaseMessagingDelegate.java#L51-L55

and

https://github.com/expo/expo/blob/08818849f48b23da2b04cfc0f6641aae733ce877/android/expoview/src/main/java/host/exp/exponent/experience/ExperienceActivity.java#L529

I have reverted offending commit and left the code intact.